swesmith/bug_gen/procedural/rust/operations.py

```diff
@@ -0,0 +1,448 @@
+import tree_sitter_rust as tsrs
+
+from swesmith.bug_gen.procedural.base import CommonPMs
+from swesmith.bug_gen.procedural.rust.base import RustProceduralModifier
+from swesmith.constants import BugRewrite, CodeEntity
+from tree_sitter import Language, Parser
+
+RUST_LANGUAGE = Language(tsrs.language())
+
+FLIPPED_OPERATORS = {
+    "+": "-",
+    "-": "+",
+    "*": "/",
+    "/": "*",
+    "%": "*",
+    "<<": ">>",
+    ">>": "<<",
+    "&": "|",
+    "|": "&",
+    "^": "&",
+    "==": "!=",
+    "!=": "==",
+    "<": ">",
+    "<=": ">=",
+    ">": "<",
+    ">=": "<=",
+    "&&": "||",
+    "||": "&&",
+}
+
+# Operator groups for systematic changes
+ARITHMETIC_OPS = ["+", "-", "*", "/", "%"]
+BITWISE_OPS = ["&", "|", "^", "<<", ">>"]
+COMPARISON_OPS = ["==", "!=", "<", "<=", ">", ">="]
+LOGICAL_OPS = ["&&", "||"]
+
+
+class OperationChangeModifier(RustProceduralModifier):
+    explanation: str = CommonPMs.OPERATION_CHANGE.explanation
+    name: str = CommonPMs.OPERATION_CHANGE.name
+    conditions: list = CommonPMs.OPERATION_CHANGE.conditions
+
+    def modify(self, code_entity: CodeEntity) -> BugRewrite:
+        """Apply operation changes to Rust binary expressions."""
+        if not self.flip():
+            return None
+
+        parser = Parser(RUST_LANGUAGE)
+        tree = parser.parse(bytes(code_entity.src_code, "utf8"))
+
+        modified_code = self._change_operations(code_entity.src_code, tree.root_node)
+
+        if modified_code == code_entity.src_code:
+            return None
+
+        return BugRewrite(
+            rewrite=modified_code,
+            explanation=self.explanation,
+            strategy=self.name,
+        )
+
+    def _change_operations(self, source_code: str, node) -> str:
+        """Recursively find and change binary operations."""
+        modifications = []
+
+        def collect_binary_ops(n):
+            if n.type == "binary_expression":
+                operator_node = None
+                for child in n.children:
+                    if child.type in [
+                        "+",
```

Comment:
This set of operators is repeated several times in this file. It may be nice to reduce the duplication here.